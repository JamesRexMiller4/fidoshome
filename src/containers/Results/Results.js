import React from 'react';
import './Results.scss';
import Card from '../Card/Card';
import { connect } from 'react-redux';
import { addAnimals } from '../../actions/index';
import NoPhoto from '../../images/No_picture_available.png';

const Results = ({ animals }) => {
  console.log(animals)
  let cards, photos;
  if (animals.length > 1) {
    cards = animals.map((animal, index) => {
      if (animal.photos.length < 1) {
        photos=NoPhoto
      } else {
        photos= animal.photos[0].medium
      }
      return (<Card
        key={index + 1}
        id={animal.id}
        age={animal.age}
        name={animal.name}
        img={photos}
        address={animal.contact.address.city}
      />)
    })
    let button = (<button type='button' className='results-btn'>More Results
    <span aria-label='dog-emoji' role='img'>üê∂</span></button>)
    cards.push(button)
  } else {
    cards = [];
  }

  return ( 
    <section className='results-section'>
      {cards}
    </section>
  );
}

const mapStateToProps = state => ({
  animals: state.animals
})

const mapDispatchToProps = dispatch => ({
  addAnimals: animals => dispatch(addAnimals(animals))
})

export default connect(mapStateToProps, mapDispatchToProps)(Results);