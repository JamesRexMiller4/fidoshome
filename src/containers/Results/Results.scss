.results-section {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 20px;
  width: 100%;
  height: 80vh;
  background-color: rgb(235, 233, 233);
  margin: 30px;
  margin-left: -10px;
  border: 5px solid;
  border-color: rgba(87, 37, 186, 0.844);
  overflow: scroll;
}