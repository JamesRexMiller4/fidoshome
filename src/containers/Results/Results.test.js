import React from 'react';
import {shallow} from 'enzyme';
import { Results, mapStateToProps, mapDispatchToProps } from './Results';
import VeryGoodBoy from '../../images/verygoodboy.PNG';

describe('Results Container', () => {
  let mockData;
  beforeEach(() => {
    mockData = [{
      photos: [{medium: VeryGoodBoy}],
      id: 4, 
      name: 'Fido',
      age: 'Older than time itself',
      contact: {address: {city: 'Venice'}}
    }]
  });

  it('should match the snapshot with no data passed through', () => {
      const wrapper = shallow(<Results 
        animals={[]}
        />)
      expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot with data passed through', () => {
      const wrapper = shallow(<Results 
      animals={mockData}
      />);
      expect(wrapper).toMatchSnapshot();
  });

  it('should render a "no photo available" picture if the data does not have a photo', () => {
    const wrapper = shallow(<Results 
      animals={[{
        photos: [],
        id: 4, 
        name: 'Fido',
        age: 'Older than time itself',
        contact: {address: {city: 'Venice'}}
      }]}
      />);
      expect(wrapper).toMatchSnapshot();
  });
    describe('mapStateToProps', () => {
      it('should return an object with animals', () => {
        const mockState = {animals: mockData};
        const expected = {animals: mockData};
        const mappedProps = mapStateToProps(mockState);
        expect(mappedProps).toEqual(expected);
      });
    });
});