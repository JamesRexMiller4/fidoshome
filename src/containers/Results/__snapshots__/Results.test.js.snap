// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Container should match the snapshot with data passed through 1`] = `
<section
  className="results-section"
>
  <Card
    address="Venice"
    age="Older than time itself"
    id={4}
    img="verygoodboy.PNG"
    key="1"
    name="Fido"
  />
  <button
    className="results-btn"
    key="42"
    type="button"
  >
    More Results
    <span
      aria-label="dog-emoji"
      role="img"
    >
      🐶
    </span>
  </button>
</section>
`;

exports[`Results Container should match the snapshot with no data passed through 1`] = `
<section
  className="results-section"
/>
`;

exports[`Results Container should render a "no photo available" picture if the data does not have a photo 1`] = `
<section
  className="results-section"
>
  <Card
    address="Venice"
    age="Older than time itself"
    id={4}
    img="No_picture_available.png"
    key="1"
    name="Fido"
  />
  <button
    className="results-btn"
    key="42"
    type="button"
  >
    More Results
    <span
      aria-label="dog-emoji"
      role="img"
    >
      🐶
    </span>
  </button>
</section>
`;
